### ISSUE TITLE
Enum types generation issue

### ISSUE THREAD

**Issue by NagorniyAleksandr** [tinnou/graphql-codegen#2]
Hi! 
There are some small issue with generating Enum types.

1. Class name should starts from the Upper letter.
2. We have to set some value to each line, because we haven't default constructor. 
3. We should have get method for value

Now our generated code is looks like:
```
public enum jump {

    START, 
    END

    private String value;

    jump (String value) {
    this.value = value;
    }

    @Override
    public String toString() {
        return value;
    }
}
```

but. the good code would be that:
```
public enum **J**ump {

    START**("start")**, 
    END**("end");**

    private String value;

    jump (String value) {
    this.value = value;
    }

    **public String getValue() {
        return value;
    }**

    @Override
    public String toString() {
        return value;
    }
}
```

for example(or for testing), here is part of schema:
```json
{
  "kind": "ENUM",
  "name": "game",
  "description": "Valid field values for game",
  "fields": null,
  "inputFields": null,
  "interfaces": null,
  "enumValues": [
    {
      "name": "start",
      "description": null,
      "isDeprecated": false,
      "deprecationReason": null
    },
    {
      "name": "end",
      "description": null,
      "isDeprecated": false,
      "deprecationReason": null
    }
  ],
  "possibleTypes": null
}
```

I've already fixed this issue, so I can push code with fixes to some branch if you provide me permissions.

**Comment by tinnou** [comment]
Thank you for your feedback and I welcome contributions. Can you create a [pull request](https://help.github.com/articles/creating-a-pull-request/) with your proposed changes and a few tests covering them? You can probably modify the [JavaEnumTest](https://github.com/tinnou/graphql-codegen/blob/master/src/test/groovy/graphql_codegen/builder/java/JavaEnumTest.groovy) to use a lowercase enum name to cover the use case you found.

**Comment by tinnou** [comment]
Fixed by #3
